Tests/webcomponents/js/ez-universal-discovery.js

```diff
@@ -0,0 +1,440 @@
+                assert.isFunction(params.isSelectable);
+                done(false, 'whatever', view);
+            });
+            document.dispatchEvent(new CustomEvent('ez:yui-app:ready'));
+        });
+
+        it('should handle loading error', function () {
+            const errorMsg = 'error message';
+
+            renderViewStub = sinon.stub(window.eZ.YUI.app, 'renderSideView', function (View, Service, params, done) {
+                done(new Error(errorMsg));
+            });
+            document.dispatchEvent(new CustomEvent('ez:yui-app:ready'));
+
+            assert.equal(
+                universalDiscovery.innerHTML,
+                errorMsg,
+                '<ez-universal-discovery> should contain the error message'
+            );
+        });
+
+        it('should add the ez-js-standard-form class', function () {
+            const view = getViewMock(container);
+
+            renderViewStub = sinon.stub(window.eZ.YUI.app, 'renderSideView', function (View, Service, params, done) {
+                done(false, 'whatever', view);
+            });
+            document.dispatchEvent(new CustomEvent('ez:yui-app:ready'));
+            assert.isTrue(universalDiscovery.classList.contains('ez-js-standard-form'));
+        });
+
+        describe('events', function () {
+            describe('`ez:cancel`', function () {
+                it('should be dispatched', function() {
+                    let isDispatched = false;
+
+                    universalDiscovery.addEventListener('ez:cancel', function (e) {
+                        assert.isFalse(e.bubbles);
+                        isDispatched = true;
+                    });
+                    runUDWEventHandler('cancelDiscoverHandler');
+                    assert.isTrue(isDispatched);
+                });
+            });
+
+            describe('`ez:select`', function () {
+                it('should be dispatched when selecting an item', function() {
+                    const selection = {location: location, content: content, contentType: contentType};
+                    let isDispatched = false;
+
+                    universalDiscovery.addEventListener('ez:select', function (e) {
+                        assertEventParam(e);
+                        isDispatched = true;
+                    });
+                    runUDWEventHandler('isSelectable', selection);
+                    assert.isTrue(isDispatched);
+                });
+
+                it('should allow to prevent the selection', function() {
+                    const selection = {location: location, content: content, contentType: contentType};
+                    const container = document.createElement('div');
+                    const view = getViewMock(container);
+
+                    universalDiscovery.addEventListener('ez:select', function (e) {
+                        e.preventDefault();
+                        assert.isTrue(e.cancelable);
+                        return false;
```

Comment:
that's confusing, this should not be there, `e.preventDefault()` is enough to prevent the event